src/store/theme.ts

```diff
@@ -1,26 +1,82 @@
-import type { Theme } from '@lib/types/theme';
+import { DARK_THEME, LIGHT_THEME } from '@lib/consts/theme';
+import type { Theme, ThemeName } from '@lib/types/theme';
 import { writable } from 'svelte/store';
 
-const light: Theme = {
-    fg: '#000',
-    bg: '#fff',
-    text: '#000',
-    icon: '#000',
-};
+export let themes: Theme[] = [
+    {
+        name: LIGHT_THEME,
+        vars: {
+            '--primary-color': '#00ff9a',
+            '--secondary-color': '#000',
+            '--main-fg': '#000',
+            '--main-fg-inverse': '#fff',
+            '--main-bg': '#fff',
+            '--main-bg-inverse': '#000',
+            '--main-text': '#000',
+            '--main-text-inverse': '#fff',
+            '--button-text': '#000',
+            '--button-text-hover': '#fff',
+            '--button-text-inverse': '#fff',
+            '--input-text': '#000',
+            '--input-bg': '#ddd',
+            '--input-bg-hover': '#ccc',
+            '--input-bg-focus': '#ccc',
+            '--input-bg-inverse': '#222',
+            '--input-border': '3px solid #888',
+            '--input-border-focus': '3px solid #000',
+            '--button-bg': 'transparent',
+            '--button-bg-hover': '#000',
+            '--button-bg-inverse': '#fff',
+            '--shadow': '0 0 1rem rgba(0, 0, 0, 0.5)',
+            '--border': '1px solid #000',
+            '--border-color': '#000',
+            '--border-radius': '0.3rem',
+        },
+    },
+    {
+        name: DARK_THEME,
+        vars: {
+            '--primary-color': '#00ff9a',
+            '--secondary-color': '#fff',
+            '--main-fg': '#fff',
+            '--main-fg-inverse': '#222',
+            '--main-bg': '#222',
+            '--main-bg-inverse': '#fff',
+            '--main-text': '#fff',
+            '--main-text-inverse': '#000',
+            '--button-text': '#fff',
+            '--button-text-hover': '#000',
+            '--button-text-inverse': '#fff',
+            '--input-text': '#fff',
+            '--input-bg': '#333',
+            '--input-bg-hover': '#222',
+            '--input-bg-focus': '#222',
+            '--input-bg-inverse': '#ccc',
+            '--input-border': '3px solid #ddd',
+            '--input-border-focus': '3px solid #fff',
+            '--button-bg': 'transparent',
+            '--button-bg-hover': '#fff',
+            '--button-bg-inverse': '#000',
+            '--shadow': '0 0 0.5rem #fff',
+            '--border': '1px solid #fff',
+            '--border-color': '#fff',
+            '--border-radius': '0.5rem',
+        },
+    },
+];
 
-const dark: Theme = {
-    fg: '#fff',
-    bg: '#000',
-    text: '#fff',
-    icon: '#fff',
-};
+const getTheme = (name: ThemeName): Theme =>
+    themes.find((t) => t.name === name);
 
-export const theme = writable<Theme>(light);
+export const themeStore = writable<Theme>(getTheme(LIGHT_THEME));
```

Comment:
Let's base the initial theme on the user's OS theme
```suggestion
function getOsTheme(): ThemeName {
    const prefersDarkMq = window.matchMedia('(prefers-color-scheme: dark)');
    return prefersDarkMq.matches ? 'DARK_THEME' : 'LIGHT_THEME';
}

export const themeStore = writable<Theme>(getTheme(getOsTheme()));
```